sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleLastValueAggregator.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics.aggregator;
+
+import com.google.common.util.concurrent.AtomicDouble;
+import io.opentelemetry.sdk.metrics.data.MetricData.DoublePoint;
+import io.opentelemetry.sdk.metrics.data.MetricData.Point;
+import java.util.Map;
+
+/**
+ * Aggregator that aggregates recorded values by storing the last recorded value.
+ *
+ * <p>Limitations:
+ *
+ * <ul>
+ *   <li>The current implementation does not store a time when the value was recorded, so merging
+ *       multiple LastValueAggregators will not preserve the ordering of records. This is not a
+ *       problem because LastValueAggregator is currently only available for Observers which record
+ *       all values once.
+ *   <li>The current implementation does not properly reset the current value, it should use a
+ *       {@code null} value instead of 0. This is not a problem because LastValueAggregator is
+ *       currently only available for Observers which do not reuse reset instances.
```

Comment:
Is it possible that an aggregator is asked to report data *before* the observer has had a chance to report, hence leading to a zero value recorded? The lifecycle in this is unclear to me. I'm also ok addressing this in a follow-up PR.

Reply:
That behavior is not possible.

Reply:
A design doc that explains this would be super helpful.  :)

Reply:
Not sure what to do now :) do you mind clarify what you want to see?

Reply:
Done. No more chances :)

Reply:
cool. thanks. Still would love to see an overall design documentation about how this metrics system works, for future us. The code is pretty complex, and documentation will pay dividends in the long run.